### PR TITLE
Add "Redirecting" page in the Routing section

### PR DESCRIPTION
**Pull request by delbaoliveira**
New page outlining all options for handling redirects in Next.js, from the simplest use cases (e.g. `redirect()` in a Server Action), to more advanced cases (e.g. bloom filters and middleware). 

- [x] `redirect`
- [x] `permanentRedirect`
- [x] `useRouter`
- [x] `next.config.js`
- [x] Middleware
- [x] Handling redirects at scale
  - [x] Edge config
  - [x] Bloom filter
 

**Bloom Filter Example:** https://github.com/vercel/examples/pull/857

### CODE REVIEW

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+      console.error(error)
+    }
+  }
+
+  // No redirect found, continue the request without redirecting
+  return NextResponse.next()
+}
+```
+
+The in the Route Handler:
```
Comment (delbaoliveira) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:449]:
```suggestion
Then, in the Route Handler:
```

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+  // Fetch redirect data from database
+  const redirectData = await get(pathname)
+
+  if (redirectData && typeof redirectData === 'string') {
+    const redirectEntry: RedirectEntry = JSON.parse(redirectData)
+
+    // Determine the status code
+    const statusCode = redirectEntry.permanent ? 308 : 307
+
+    // Redirect to the destination
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:307]:
```suggestion
  const pathname = request.nextUrl.pathname
  const redirectData = await get(pathname)

  if (redirectData && typeof redirectData === 'string') {
    const redirectEntry: RedirectEntry = JSON.parse(redirectData)
    const statusCode = redirectEntry.permanent ? 308 : 307
```

I think the code is explanatory enough!

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+  // Fetch redirect data from database
+  const redirectData = await get(pathname)
+
+  if (redirectData) {
+    const redirectEntry = JSON.parse(redirectData)
+
+    // Determine the status code
+    const statusCode = redirectEntry.permanent ? 308 : 307
+
+    // Redirect to the destination
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:333]:
```suggestion
  const pathname = request.nextUrl.pathname
  const redirectData = await get(pathname)

  if (redirectData) {
    const redirectEntry = JSON.parse(redirectData)
    const statusCode = redirectEntry.permanent ? 308 : 307
```

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+  // No redirect found, continue without redirecting
+  return NextResponse.next()
+}
+```
+
+### 2. Optimizing data lookup performance
+
+Reading a large dataset for every incoming request can be slow and expensive. There are two ways you can optimize data lookup performance:
+
+- Use a database that is optimized for fast reads, such as [Edge Config](https://vercel.com/docs/storage/edge-config/get-started) or [Redis](https://redis.io/).
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:346]:
```suggestion
- Use a database that is optimized for fast reads, such as [Vercel Edge Config](https://vercel.com/docs/storage/edge-config/get-started?utm_source=next-site&utm_medium=docs&utm_campaign=next-website) or [Redis](https://vercel.com/docs/storage/vercel-kv?utm_source=next-site&utm_medium=docs&utm_campaign=next-website).
```

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+- Use a data lookup strategy such as a [Bloom filter](https://en.wikipedia.org/wiki/Bloom_filter) to efficiently check if a redirect exists **before** reading the larger redirects file.
+
+Considering the previous example, you can import a generated Bloom Filter file into Middleware, then, check if the incoming request's pathname exists in the Bloom Filter.
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:349]:
```suggestion
Considering the previous example, you can import a generated bloom filter file into Middleware, then, check if the incoming request pathname exists in the bloom filter.
```

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+  }
+
+  // Return the redirect entry
+  return NextResponse.json(redirect)
+}
+```
+
+> **Good to know:**
+>
+> - To generate a Bloom filter, you can use a library like [bloom-filters](https://www.npmjs.com/package/bloom-filters).
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:504]:
```suggestion
> - To generate a bloom filter, you can use a library like [`bloom-filters`](https://www.npmjs.com/package/bloom-filters).
```

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+> - You should validate requests made to your Route Handler to prevent malicious requests.
+> - On Vercel, Middleware runs on the [Edge Network](https://vercel.com/docs/edge-network/overview), check the Vercel or your platform's [documentation for limits](https://vercel.com/docs/storage/edge-config/edge-config-limits).
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:506]:
```suggestion
```

I'm not sure this one is needed, I would say cut.

**Review by leerob — APPROVED** [review]

File: docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx [hunk 8]
```diff
@@ -0,0 +1,491 @@
...
+> - `redirect` returns a 307 (Temporary Redirect) status code by default. When used in a Server Action, it returns a 303 (See Other), which is commonly used for redirecting to a success page as a result of a POST request.
+> - `redirect` internally throws an error so should be called outside of `try/catch` blocks.
+> - `redirect` can be called in Client Components during the rendering process but not in event handlers. You can use the [`useRouter` hook](#userouter-hook) instead.
+> - If you'd like to redirect before the render process, use [`next.config.js`](#redirects-in-nextconfigjs) or [Middleware](#nextresponseredirect-in-middleware).
+
+See the [`redirect` API reference](/docs/app/api-reference/functions/redirect) for more information.
+
+## `permanentRedirect` function
+
+The `permanentRedirect` function allows you to **permanently** redirect the user to another URL. You can call `redirect` in [Server Components](/docs/app/building-your-application/rendering/server-components), [Route Handlers](/docs/app/building-your-application/routing/route-handlers), and [Server Actions](/docs/app/building-your-application/data-fetching/server-actions-and-mutations).
```
Comment (A7med3bdulBaset) [inline docs/02-app/01-building-your-application/01-routing/06-redirecting.mdx:75]:
```suggestion 
The `permanentRedirect` function allows you to **permanently** redirect the user to another URL. You can call `permanentRedirect` in [Server Components](/docs/app/building-your-application/rendering/server-components), [Route Handlers](/docs/app/building-your-application/routing/route-handlers), and [Server Actions](/docs/app/building-your-application/data-fetching/server-actions-and-mutations).
```

opened #60583

**Pull request merged**